### DISCUSSION TITLE
[@types/node] "Types of property 'createConnection' are incompatible" error on typescript 5.1 with `exactOptionalPropertyTypes` enabled

### DISCUSSION THREAD

**Discussion by DetachHead** [DefinitelyTyped/DefinitelyTyped#64613] [Issues with a @types package] [answered]
```
node_modules/@types/node/http2.d.ts:1058:22 - error TS2430: Interface 'ClientSessionOptions' incorrectly extends interface 'SessionOptions'.
  Types of property 'createConnection' are incompatible.
    Type '((authority: URL, option: SessionOptions) => Duplex) | undefined' is not assignable to type '(authority: URL, option: SessionOptions) => Duplex'.
      Type 'undefined' is not assignable to type '(authority: URL, option: SessionOptions) => Duplex'.

1058     export interface ClientSessionOptions extends SessionOptions {
                          ~~~~~~~~~~~~~~~~~~~~


Found 1 error in node_modules/@types/node/http2.d.ts:1058
```
typescript 5.1.0-dev.20230306, [playground](https://www.typescriptlang.org/play?noUncheckedIndexedAccess=true&allowUnreachableCode=true&allowUnusedLabels=true&noUnusedLocals=true&target=99&jsx=0&isolatedModules=true&useUnknownInCatchVariables=true&exactOptionalPropertyTypes=true&noFallthroughCasesInSwitch=true&noImplicitOverride=true&noPropertyAccessFromIndexSignature=true&suppressImplicitAnyIndexErrors=false&ts=5.1.0-dev.20230306#code/KYDwDg9gTgLgBASwHY2FAZgQwMbDgZWAGciEIkB5MGMpIuAbwCg5W5sphNUBhcpYNhrkA-AApMAVxgALaAhgBPAFxwiMKMgDmAGjgRqtVYRK0qwugEpV6zUi0BuJgF8moSLEQo0WXHB4ANgjAKCak5Oa09KCoSAAm9GFmhuT0zGzsnNzAfEgCQrQiqmIS0nKaSjYa2noGFsbE4ZQpVnAAvAB8atX2lnAAPnCS8cDoyMBxTs5AA)

[a comment by an automated account is not shown]

**Reply by jacobbogers** [reply]
Hi, I am using version Version 5.2.2 and I am getting this exact error,  why?

**Reply by DetachHead** [reply]
are you using the latest version of `@types/node`?

**Comment by Semigradsky** [comment, marked as answer]
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64614
